src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -608,6 +608,62 @@ public void RegisterSymbolAction(DiagnosticAnalyzer analyzer, Action<SymbolAnaly
             SymbolAnalyzerAction analyzerAction = new SymbolAnalyzerAction(action, symbolKinds, analyzer);
             this.GetOrCreateAnalyzerActions(analyzer).AddSymbolAction(analyzerAction);
             _symbolActions = _symbolActions.Add(analyzerAction);
+
+            // The SymbolAnalyzerAction does not handle SymbolKind.Local or SymbolKind.Parameter
+            // because the compiler does not make CompilationEvents for them. As a workaround,
+            // handle them specially by registering further OperationActions (for 
+            // VariableDeclarations) or SymbolActions (for Methods) and utilize the results of
+            // those to construct the necessary SymbolAnalysisContexts.
+
+            if (symbolKinds.Contains(SymbolKind.Local))
+            {
+                RegisterOperationAction(analyzer, opContext =>
```

Comment:
How is this preferable over actually asking the analyzer authors to register these operation actions? I am hesitant on ourselves registering new actions, as this will also skew up telemetry (for example, how many operation actions were registered).

If we do want to support these new symbol kinds, then the correct approach might be to just execute these directly in http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalyzerExecutor.cs,325